### PR TITLE
Add docs for --x-cmake-debug and --x-cmake-configure-debug

### PR DESCRIPTION
**Pull request by BillyONeal**
See https://github.com/microsoft/vcpkg-tool/pull/1173

### CODE REVIEW

**Comment by Neumann-A** [comment]
Opened https://github.com/BillyONeal/vcpkg-docs/pull/1 to add additional info.

**Comment by AugP** [comment]
Reviewed and talked to @BillyONeal offline about this PR. Billy once you incorporate @Neumann-A's additional feedback we can proceed. If the tool PR associated with this doc PR is coming in the next release we can merge this, otherwise we can move it to draft until we get close to the date.

**Comment by BillyONeal** [comment]
> Opened [BillyONeal#1](https://github.com/BillyONeal/vcpkg-docs/pull/1) to add additional info.

Thank you for both writing good docs and smashing them into mine!

File: vcpkg/commands/common-options.md [hunk 1]
```diff
@@ -29,6 +29,45 @@ Specifies the temporary path to store intermediate build files, such as objects
...
+Note: Unlike `--x-cmake-debug` there will be no output indicating that the build is waiting for a debugger to connect. The output will simply stall at `Configuring <triplet>`.
+
+Note: Although all ports should be calling CMake inside of the portfile to obtain compile flags from the CMake toolchain. This command is mainly useful for ports actually using CMake as their build system.
+
+### <a name="cmake-debug"></a> `--x-cmake-debug=<debugger-pipe>[;semicolon;port;list]`
+
+[!INCLUDE [experimental](../../includes/experimental.md)]
+
+Enables CMake's [`--debugger`](https://cmake.org/cmake/help/latest/manual/cmake.1.html#cmdoption-cmake-debugger) while
+running directly-invoked CMake scripts like triplet `.cmake`s or `portfile.cmake`s.
```
Comment (vicroms) [inline vcpkg/commands/common-options.md:60]:
```suggestion
running directly-invoked CMake scripts like triplet files (for example, `x64-windows.cmake`) or `portfile.cmake` files.
```

> `.cmake`s or `portfile.cmake`s 

Mixing code-style with regular font looks strange when rendered in the Learn platform. I'm also concerned that it won't localize cleanly and might cause issues with screen readers.

**Comment by vicroms** [comment]
There is a `next-release` branch which you can target instead for PRs that are awaiting a tool release.

**Pull request merged**